### PR TITLE
Fix create containerapp/containerappjob with msi issue

### PR DESCRIPTION
**Pull request by njuCZ**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/containerapp/HISTORY.rst [hunk 1]
```diff
@@ -4,6 +4,7 @@ Release History
 ===============
 upcoming
 ++++++
+* Fix create container app / contianer app job with msi issue
```
Comment (Greedygre) [inline src/containerapp/HISTORY.rst:7]: 'az containerapp create`: Fix xxxx
'az containerapp job create': Fix xxxx

File: src/containerapp/azext_containerapp/_utils.py [hunk 2]
```diff
@@ -749,13 +749,25 @@ def is_registry_msi_system_environment(identity):
...
 def env_has_managed_identity(cmd, resource_group_name, env_name, identity):
     identity = identity.lower()
-    # caller should handle exception
-    managed_env_info = ManagedEnvironmentPreviewClient.show(cmd=cmd, resource_group_name=resource_group_name, name=env_name)
-    if is_registry_msi_system(identity) and managed_env_info["identity"]["type"].__contains__("SystemAssigned"):
+    
+    try:
+        managed_env_info = ManagedEnvironmentPreviewClient.show(cmd=cmd, resource_group_name=resource_group_name, name=env_name)
+    except:
+        return False
```
Comment (Greedygre) [inline src/containerapp/azext_containerapp/_utils.py:756]: We should handle exception if not expect.

**Comment by Greedygre** [comment]
Please fix the style.
```

ERROR: ************* Module azext_containerapp._utils
src/containerapp/azext_containerapp/_utils.py:752:0: C0303: Trailing whitespace (trailing-whitespace)
src/containerapp/azext_containerapp/_utils.py:757:0: C0303: Trailing whitespace (trailing-whitespace)
```

**Review by Greedygre — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

File: src/containerapp/azext_containerapp/_utils.py [hunk 3]
```diff
@@ -749,13 +749,25 @@ def is_registry_msi_system_environment(identity):
 
 def env_has_managed_identity(cmd, resource_group_name, env_name, identity):
     identity = identity.lower()
-    # caller should handle exception
-    managed_env_info = ManagedEnvironmentPreviewClient.show(cmd=cmd, resource_group_name=resource_group_name, name=env_name)
-    if is_registry_msi_system(identity) and managed_env_info["identity"]["type"].__contains__("SystemAssigned"):
+
+    try:
+        managed_env_info = ManagedEnvironmentPreviewClient.show(cmd=cmd, resource_group_name=resource_group_name, name=env_name)
+    except Exception as e:
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/_utils.py:755]: only handle 404

File: src/containerapp/azext_containerapp/tests/latest/test_containerappjob_with_identity.py [hunk 4]
```diff
@@ -258,4 +257,55 @@ def test_containerappjob_identity_registry(self, resource_group):
...
+        roleAssignmentName = self.create_guid()
+        self.cmd(f'role assignment create --role acrpull --assignee {identity_json["principalId"]} --scope {acr_id} --name {roleAssignmentName}')
+        # upload image
+        self.cmd(f'acr import -n {acr} --source {image_source}')
+
+        # wait for role assignment take effect
+        time.sleep(30)
+
+        # use msi to pull image
+        self.cmd(f'containerapp job create -g {resource_group} -n {job} --environment {env_id} --trigger-type manual --replica-timeout 5 --replica-retry-limit 2 --replica-completion-count 1 --parallelism 1 --image {image_name} --registry-server {acr}.azurecr.io --registry-identity {user_identity_id}')
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/tests/latest/test_containerappjob_with_identity.py:304]: please delete job and env

**Pull request merged**